proposals/20231220-features-adoption-and-deprecation.md

```diff
@@ -0,0 +1,207 @@
+6. Any Pull Request introducing a deprecation notice must be labeled and include a note in the format `DEPRECATION NOTICE: <description>`.
+7. Any Pull Request introducing a breaking change due to the end of the deprecation notice period must be labeled and include a note in the format `BREAKING CHANGE:`. 
+   - It is also recommended for code commits that introduce a breaking change to follow the related [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/#specification).
+
+The minimum deprecation period length depends on the specific subsystem. If changes span multiple areas, the area with the most extended deprecation period is assumed.
+
+**Deprecation Periods (up to Falco 0.x)**
+
+| Area           | Stable | Incubating |
+| -------------- | ------ | ---------- |
+| Behaviors      | 1      | 1          |
+| Output/Alerts  | 1      | 1          |
+| CLI/Config     | 1      | 0          |
+| Rules          | 1      | 0          |
+| Subsystem APIs | 0      | 0          |
+
+**Deprecation Periods (since Falco 1.0 onward)**
+
+| Area           | Stable | Incubating |
+| -------------- | ------ | ---------- |
+| Behaviors      | 2      | 1          |
+| Output/Alerts  | 2      | 1          |
+| CLI/Config     | 2      | 1          |
+| Rules          | 2      | 1          |
+| Subsystem APIs | 1      | 1          |
+
+### Exceptions
+
+- Subsystems or subcomponents may have additional criteria and exceptions (for instance, [the rules maturity framework](https://github.com/falcosecurity/rules/blob/main/CONTRIBUTING.md#rules-maturity-framework)). Stated other criteria and exceptions must not directly affect the main Falco distribution (e.g., *falcoctl* can have a different release cycle and different policies; however, if Falco relies on a specific *falcoctl* feature, that *falcoctl* feature adoption and deprecation must be strictly compatible with the rules described in this proposal).
+- Internal APIs are out of scope for this policy. Their adoption models and deprecation policies might be regulated separately.
+- Different parties may provide plugins, and each plugin may have a different maturity level. Only those plugins officially maintained by The Falco Project and identified as "core" for Falco are in scope for this policy; all others are excluded.
+- Any other exceptions to the rules provided by this policy require a formal core maintainer majority vote.
+
+### Versioning
+
+Regarding the above policies, component versioning must adhere to [Semantic Versioning 2.0.0](https://semver.org/). However, in the context of Falco core components, the scope extends beyond the strict API definition and includes any user-facing changes.
+
+Thus, given a version number `MAJOR.MINOR.PATCH` increment the:
+
+- *MAJOR* version when the deprecation period ends, thus introducing incompatible user-facing or API changes.
```

Comment:
```suggestion
- *MAJOR* version when the deprecation period of one or more _stable_ features ends, thus introducing incompatible user-facing or API changes.
```

This edit would prevent us from bumping the MAJOR when introducing a breaking change in incubating/stable features. Otherwise, the MAJOR version will increase very quickly and we probably don't want that.
wdyt?